src/mem/qos/QoSMemSinkCtrl.py

```diff
@@ -46,9 +45,9 @@ class QoSMemSinkCtrl(QoSMemCtrl):
     cxx_class = "gem5::memory::qos::MemSinkCtrl"
     port = ResponsePort("Response ports")
 
-    interface = Param.QoSMemSinkInterface(
-        QoSMemSinkInterface(), "Interface to memory"
-    )
```

Comment:
Correct me if I'm wrong, but I'm pretty sure the code below would work.

```
interface = Param.QoSMemSinkInterface("Interface to memory")
```

This is not a circular dependency because it is in `params` not `m5.objects.QoSMemSinkInterface`.

Generally, problems like this are why we should not instantiate SimObjects as default values for parameters in the SimObject description files.

Let me know if I'm wrong, though.

Reply:
Hi @powerjg, I tested your suggestion and it indeed worked - I've updated the commit to feature it. 

I thought it was originally a circular dependency problem as:

1. `QoSMemSinkInterface.controller()` was instantiating `QoSMemSinkCtrl` (so `QoSMemSinkInterface.py` needed to import `QoSMemSinkCtrl.py`)
2. `QoSMemSinkCtrl()` was instantiating `QoSMemSinkInterface` as a default param (so `QoSMemSinkCtrl.py` needed to import `QoSMemSinkInterface.py`)

I agree with the reasoning that SimObjects as default values for parameters can be problematic, especially when they are coupled like in this case.